Feature/sonar qube workflow integration

RamonKea-inh opened the pull request.

RamonKea-inh committed:
Added SonarQube scan to pr validation

RamonKea-inh committed:
Fixed projectKey value now being lower case

It remains to be seen if this is right or that you would need the name off the repo as it now grabs the name off the org?

RamonKea-inh committed:
Merge branch 'development' into feature/SonarQube-workflow-integration

RamonKea-inh committed:
added a quality gate check to pr validation

RamonKea-inh committed:
added SonarQube scans to dev and prod workflows

RamonKea-inh committed:
removed unnecessary linting code

RamonKea-inh committed:
Replaced unnecessary properties with env variables

RamonKea-inh committed:
changed SONAR_ORG to SONAR_PROJECT_KEY

RamonKea-inh committed:
changed another SONAR_ORG to SONAR_PROJECT_KEY

RamonKea-inh committed:
changed missed env variables and readded a property for SonarQube

RamonKea-inh committed:
Added a JDK set up so that the workflow is also running java 17 and not 11 which causes an error for the sonar dependency

RamonKea-inh committed:
Added fetch-depth 0 to all workflows allowing SonarCloud to access full git blame information

RamonKea-inh committed:
fixed a warning off a deprecated property

RamonKea-inh committed:
Made the production deployment java JDK 17 and gave the sonar plugin the newest version

RamonKea-inh commented:
The deployment only fails because the overall testing code coverage is to low

CyberAve2100 reviewed (dismissed):
De workflows zien er erg netjes uit, goed bezig! 
Misschien kunnen we wat issues aan de board toevoegen om te zorgen dat deze tests wel een succes worden

RamonKea-inh committed:
Added a configuration file for sonarCloud

RamonKea-inh committed:
Merge branch 'development' into feature/SonarQube-workflow-integration

SemPlaatsman committed:
Restructured SonarQube to SonarCloud

SemPlaatsman committed:
Added jacoco profile and deleted bad test (tayam)

SemPlaatsman committed:
Improved repo name

SemPlaatsman committed:
Update pom.xml

SemPlaatsman committed:
Update pr-validation.yml

RamonKea-inh committed:
Merge branch 'development' into feature/SonarQube-workflow-integration

RamonKea-inh committed:
Revert "Update pr-validation.yml"

This reverts commit 868291d7a226b61cf5adee9b51b52b4c5a79d654.

RamonKea-inh committed:
Revert "Added jacoco profile and deleted bad test (tayam)"

This reverts commit 995823bbc125f709348c8dcf5b975afd307e504c.

RamonKea-inh committed:
reversed some commits

Tayam-El-Jebari commented on .github/workflows/deploy-prod.yml:
why cache your SonarCloud packages and maven packages in pr-validation.yml, but not here?

Tayam-El-Jebari commented on .github/workflows/deploy-prod.yml:
```diff
@@ -18,12 +18,59 @@ jobs:
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
+          
+          # Check quality gate status
+          QUALITY_GATE_STATUS=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/qualitygates/project_status?analysisId=$ANALYSIS_ID" | \
+          jq -r '.projectStatus.status')
+          
+          echo "Quality Gate Status: $QUALITY_GATE_STATUS"
+          
+          if [ "$QUALITY_GATE_STATUS" != "OK" ]; then
+            echo "❌ Quality Gate Failed"
+            exit 1
+          fi
+
```
This works in theory, but when i searched online i found out that there exists an official github action from sonar themself which does the quality gate checking, see:
https://github.com/SonarSource/sonarqube-quality-gate-action?tab=readme-ov-file#usage
This has more error handling and better status reporting.

Tayam-El-Jebari commented on .github/workflows/deploy-dev.yml:
```diff
@@ -28,6 +30,54 @@ jobs:
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
+
+          # Check quality gate status
+          QUALITY_GATE_STATUS=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/qualitygates/project_status?analysisId=$ANALYSIS_ID" | \
+          jq -r '.projectStatus.status')
```
same as comment below

Tayam-El-Jebari commented on .github/workflows/deploy-prod.yml:
```diff
@@ -18,12 +18,59 @@ jobs:
+            -Dsonar.host.url=${{ secrets.SONAR_HOST_URL }} \
+            -Dsonar.token=${{ secrets.SONAR_TOKEN }}
+          
+          # Quality Gate Check
+          SONAR_CE_TASK_URL=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?componentKey=${{ secrets.SONAR_PROJECT_KEY }}")
+          
+          TASK_ID=$(echo $SONAR_CE_TASK_URL | jq -r '.task.id')
+          
+          # Wait for task to complete and get analysis ID
+          ANALYSIS_ID=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
```
same as comment below, there exists a github action for this.

RamonKea-inh committed:
removed quality gate from pr validations

RamonKea-inh committed:
removed some brackets causing a problem

RamonKea-inh committed:
changed the organization key

RamonKea-inh committed:
updated the projectkey to hopefully get the right one

RamonKea-inh committed:
changed the projectKey again

RamonKea-inh committed:
Removed SonarQube from workflows

RamonKea-inh committed:
removed cache for sonarCloud packages and added cache for Maven packages to the other workflows

Benefits of adding this cache step include:

Faster subsequent builds
Reduced network bandwidth usage
Less time spent downloading dependencies
More consistent and predictable build times

RamonKea-inh commented:
Code analyses are now done through Sonar Cloud completly out off the workflows at the cost off the code coverage quality gate, but this was to strict and not changeble in the free version so it might be an option for later.

RamonKea-inh committed:
Added instructions in the README.md file for adding a project to sonarCube Cloud

SemPlaatsman committed:
Removed duplicate/unnecessary code
